### ISSUE TITLE
12.2.10 Upgrade - Testing Conc Job - SBS CPQ File Extract Interface Inventory Items - is Failing

### ISSUE THREAD

**Issue by hamiltonmark** [Fujitsu-America/Fujitsu-Repository#15]
**Describe the bug**
Testing Conc Job - SBS CPQ File Extract Interface Inventory Items - is Failing

See uploaded Screenshots

**Responsibility Name**
Responsibility:  PBS Troubleshooter

**Navigation**
Navigation:  Submit Request > SBS CPQ File Extract Interface Inventory Items

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by hamiltonmark** [comment]
[STG Testing CPQ Interface Items.docx](https://github.com/Fujitsu-America/Fujitsu-Repository/files/6586356/STG.Testing.CPQ.Interface.Items.docx)

**Comment by vinaytalasila** [comment]
Mark,

We will look into this and keep you posted.

Thanks
Vinay Talasila

**Comment by hamiltonmark** [comment]
Thanks Vinay
There are 3 jobs failing with the same issue:

SBS CPQ File Extract Interface Inventory Items
SBS CPQ File Extract Interface BOMs
SBS CPQ File Extract Interface Price List

**Comment by vinaytalasila** [comment]
@amitjori : As per our today's meeting, could you please update this issue with reference to the directories.

**Comment by hamiltonmark** [comment]
Re-tested ... same error....
SBS_CPQ_INV_INF_CONC_SH: SBS CPQ File Extract Interface Inventory Items
+---------------------------------------------------------------------------+

Current system time is 03-JUN-2021 12:12:45

+---------------------------------------------------------------------------+

1. OOD: PINGING SBS SERVER to check if host is alive. 
1. RESULT: PING SBS INTERNAL - SUCCESSFUL. 
2. OOD: EXEC CONCSUB JOB TO GENERATE CPQ EXTRACT FILE.
Param1 APPS/apps123
Param2 1320
Param3 HAMILTONM
Param4 35388976
Param5 84
Param6
Param7
*******************************************
ERROR: CONCSUB GENERATED FILE NOT EXIST TO COPY - CPQ EXTRACT File COPY to SBS INTERNAL DB SERVER - FAILED(2). 
*******************************************
/tpqs2i/12.2/STG/fs2/EBSapps/appl/xbol/12.0.0/bin/SBS_CPQ_INV_INF_CONC_SH
Program exited with status 4

**Comment by hamiltonmark** [comment]
Amit 
You probably already know this but....
I believe this is the issue..... Program is trying to copy from $APPLCSF/tmp
but /tmp does not exist under $APPLCSF

[aptpqs2i@ssbsashebsap3 bin]$ cd $APPLCSF/tmp
bash: cd: /tpqs2i/12.2/STG/fs_ne/inst/TPQS2I_ssbsashebsap3/logs/appl/conc/tmp: No such file or directory

[aptpqs2i@ssbsashebsap3 bin]$ cd $APPLCSF
[aptpqs2i@ssbsashebsap3 conc]$ pwd
/tpqs2i/12.2/STG/fs_ne/inst/TPQS2I_ssbsashebsap3/logs/appl/conc
[aptpqs2i@ssbsashebsap3 conc]$ cd tmp
bash: cd: tmp: No such file or directory
[aptpqs2i@ssbsashebsap3 conc]$

** Previously - $APPLCSF value was      /tpqs2i/applcsf/
and the job output was stored to     $APPLCSF/tmp/
The interface output was then copied from there.

Does the $APPLCSF value need changed?

**Comment by amitjori** [comment]
Hello Mark,

Can you please test this now. We have updated the APPLCSF/tmp value in STG 12.2.10.

[aptpqs2i@ssbsashebsap3 ~]$ cd $APPLCSF/tmp
[aptpqs2i@ssbsashebsap3 tmp]$ pwd
/tpqs2i/applcsf/tmp

Regards,
Amit

**Comment by hamiltonmark** [comment]
Thanks Amit
$APPLCSF/tmp now exists so job ran successfully.
Parts.csv file was created there and shell script was able to access and copy file to SBS Server and then
onto CPQ Server for interfacing.
Below is the log file showing success.
Thanks!
Mark

[aptpqs2i@ssbsashebsap3 tmp]$ hostname
ssbsashebsap3
[aptpqs2i@ssbsashebsap3 tmp]$ cd $APPLCSF/tmp
[aptpqs2i@ssbsashebsap3 tmp]$ pwd
/tpqs2i/applcsf/tmp
[aptpqs2i@ssbsashebsap3 tmp]$ ls -ltr Parts.csv
-rwxrwxrwx 1 aptpqs2i dba 1555102 Jun  4 07:53 Parts.csv
[aptpqs2i@ssbsashebsap3 tmp]$


+---------------------------------------------------------------------------+
Business Online: Version : 12.2

Copyright (c) 1998, 2013, Oracle and/or its affiliates. All rights reserved.

SBS_CPQ_INV_INF_CONC_SH: SBS CPQ File Extract Interface Inventory Items
+---------------------------------------------------------------------------+

Current system time is 04-JUN-2021 07:52:49

+---------------------------------------------------------------------------+

1. OOD: PINGING SBS SERVER to check if host is alive. 
1. RESULT: PING SBS INTERNAL - SUCCESSFUL. 
2. OOD: EXEC CONCSUB JOB TO GENERATE CPQ EXTRACT FILE.
Param1 APPS/apps123
Param2 1320
Param3 HAMILTONM
Param4 35388369
Param5 84
Param6
Param7
2. RESULT: CPQ EXTRACT (1) DATA AND (1) XML FILE GENERATED.
3. OOD: Copy file /tpqs2i/applcsf/tmp/Parts.csv TO oci-ssh-dev.snaponbusinesssolutions.com:cpq_inv_intf/outbound/Parts.csv
3. RESULT: CPQ EXTRACT DATA FILE GENERATED - /tpqs2i/applcsf/tmp/Parts.csv 
4a. OOD: Copy file /tpqs2i/applcsf/tmp/upload_list.xml TO oci-ssh-dev.snaponbusinesssolutions.com:cpq_inv_intf/outbound/upload_list.xml
4a. RESULT: CPQ EXTRACT XML FILE GENERATED - /tpqs2i/applcsf/tmp/upload_list.xml 
5. RESULT: CPQ EXTRACT DATA AND XML FILEs copied to oci-ssh-dev.snaponbusinesssolutions.com -- SUCCESSFUL.
6. OOD: Execute Remote Script on SBS INTERNAL DB SERVER.
 
-rwxrwxrwx. 1 sbsintf sbsintf 1555102 Jun  4 07:53 /home/sbsintf/cpq_inv_intf/outbound/Parts.csv
-rwxrwxrwx. 1 sbsintf sbsintf 127 Jun  4 07:53 /home/sbsintf/cpq_inv_intf/outbound/upload_list.xml
 
7. REMOTE: Prep File (chmod, zip and cp) /home/sbsintf/cpq_inv_intf/outbound/Parts.zip upload_list.xml
  adding: Parts.csv (deflated 89%)
total 1696
-rwxrwxrwx. 1 sbsintf sbsintf  175516 Jun  4 07:53 Parts.zip
-rwxrwxrwx. 1 sbsintf sbsintf 1555102 Jun  4 07:53 Parts.csv
-rwxrwxrwx. 1 sbsintf sbsintf     127 Jun  4 07:53 upload_list.xml
7. RESULT: Prep File (chmod, zip and cp) /home/sbsintf/cpq_inv_intf/outbound/Parts.csv - /home/sbsintf/cpq_inv_intf/outbound/ - SUCCESSFUL.
 
### TEST ### TEST ### TEST SERVER ###
### TEST ### TEST ### TEST SERVER ###
### TEST ### TEST ### TEST SERVER ###
### TEST ### TEST ### TEST SERVER ###
 
sftp_user:    sbsitest1
sftp_host:    sftp2.bigmachines.com
sftp_folder:  /home/sbsitest1/upload_zip/automated/
 
8. REMOTE: Copying file /home/sbsintf/cpq_inv_intf/outbound/Parts.zip upload_list.xml TO sbsitest1@sftp2.bigmachines.com
 
Connecting to sftp2.bigmachines.com...
sftp> version
SFTP protocol version 3
sftp> cd /home/sbsitest1/upload_zip/automated/
sftp> pwd
Remote working directory: /upload_zip/automated
sftp> ls -ltr
sftp> lcd /home/sbsintf/cpq_inv_intf/outbound
sftp> lpwd
Local working directory: /home/sbsintf/cpq_inv_intf/outbound
sftp> lls
Parts.csv
Parts.zip
upload_list.xml
sftp> put Parts.zip
Uploading Parts.zip to /upload_zip/automated/Parts.zip
sftp> put upload_list.xml
Uploading upload_list.xml to /upload_zip/automated/upload_list.xml
sftp> pwd
Remote working directory: /upload_zip/automated
sftp> ls -ltr
-rwxr-xr-x    1 1431     10002         127 Jun  4 11:54 upload_list.xml
-rwxr-xr-x    1 1431     10002      175516 Jun  4 11:54 Parts.zip
sftp> quit
8. RESULT: TEST Copying file /home/sbsintf/cpq_inv_intf/outbound/Parts.zip upload_list.xml TO sbsitest1@sftp2.bigmachines.com - SUCCESSFUL
9. REMOTE: Archiving /home/sbsintf/cpq_inv_intf/outbound/Parts.csv TO /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210604AM075356.csv
 
### TEST ### ARCHIVING ###
### TEST ### ARCHIVING ###
### TEST ### ARCHIVING ###
### TEST ### ARCHIVING ###
 
Parts_20210604AM075356.csv
Parts_20210604AM075356.zip
upload_list_20210604AM075356.xml
 
9a1. REMOTE: Archiving /home/sbsintf/cpq_inv_intf/outbound/Parts.csv TO /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210604AM075356.csv
9a2. REMOTE: Archiving /home/sbsintf/cpq_inv_intf/outbound/Parts.zip TO /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210604AM075356.zip
9a3. REMOTE: Archiving /home/sbsintf/cpq_inv_intf/outbound/upload_list.xml TO /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210604AM075356.xml
 
### TEST ### HOUSEKEEPING ###
### TEST ### HOUSEKEEPING ###
### TEST ### HOUSEKEEPING ###
### TEST ### HOUSEKEEPING ###
 
9b. REMOTE: Cleanup Files - Remove /home/sbsintf/cpq_inv_intf/outbound/Parts.csv  upload_list.xml  Parts.zip
9c. REMOTE: Cleanup +14 Days Old Archive Files - Remove /oodxfer/cpq_inv_intf/outbound/archive/Parts*
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  175516 Jun  4 07:54 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210604AM075356.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1555102 Jun  4 07:54 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210604AM075356.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  170042 May 20 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210520AM053119.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1533287 May 20 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210520AM053119.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  170042 May 19 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210519AM053122.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1533287 May 19 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210519AM053122.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  170008 May 18 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210518AM053106.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1532949 May 18 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210518AM053106.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  172962 May 17 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210517AM053131.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1534683 May 17 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210517AM053131.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  170739 May 16 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210516AM053134.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1535986 May 16 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210516AM053134.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  170739 May 15 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210515AM053107.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1535986 May 15 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210515AM053107.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  172139 May 14 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210514AM053109.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1536629 May 14 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210514AM053109.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  172134 May 13 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210513AM053114.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1536629 May 13 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210513AM053114.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  172134 May 12 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210512AM053107.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1536629 May 12 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210512AM053107.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  173199 May 11 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210511AM053116.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1536629 May 11 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210511AM053116.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  172134 May 10 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210510AM053124.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1536629 May 10 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210510AM053124.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  172144 May  9 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210509AM053127.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1536629 May  9 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210509AM053127.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  172155 May  8 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210508AM053131.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1536629 May  8 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210508AM053131.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  173226 May  7 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210507AM053115.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1536629 May  7 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210507AM053115.csv
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer  170391 May  6 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210506AM053125.zip
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 1536629 May  6 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/Parts_20210506AM053125.csv
9d. REMOTE: Cleanup +14 Days Old Archive Files - Remove /oodxfer/cpq_inv_intf/outbound/archive/upload_list* 
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 Jun  4 07:54 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210604AM075356.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May 20 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210520AM053119.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May 19 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210519AM053122.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May 18 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210518AM053106.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May 17 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210517AM053131.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May 16 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210516AM053134.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May 15 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210515AM053107.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May 14 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210514AM053109.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May 13 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210513AM053114.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May 12 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210512AM053107.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May 11 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210511AM053116.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May 10 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210510AM053124.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May  9 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210509AM053127.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May  8 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210508AM053131.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May  7 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210507AM053115.xml
-rwxrwxr-x. 1 sbsintf ent_ood_dev_xfer 127 May  6 05:31 /oodxfer/cpq_inv_intf/outbound/archive/TPQS2I/upload_list_20210506AM053125.xml
10. REMOTE: Remote Archiving and Housecleaning - Completed.
11. REMOTE: Remote Job Completed.
12. REMOTE: Returning to calling Shell Script on OOD Server.
Null message body; hope that's ok
Null message body; hope that's ok
13. RESULT: Remote Script Complete.  Returned from Calling Script.
::::::::::::::
/tmp/SBS_CPQ_INV_INF_CONC_SH.prog.21679
::::::::::::::
Submitted request 35389478 for CONCURRENT PBS SBS_CPQ_LOAD_ITEMS 84 

Normal completion
14. OOD: Performing Housekeeping - Copy /tmp/SBS_CPQ_INV_INF_CONC_SH.prog.21679 Log to /oodxfer/cpq_inv_intf/logs/SBS_CPQ_INV_PKG_LOG_20210604AM075249.TPQS2I.
14. RESULT: /tmp/SBS_CPQ_INV_INF_CONC_SH.prog.21679 Copied to oci-ssh-dev.snaponbusinesssolutions.com - /oodxfer/cpq_inv_intf/logs/SBS_CPQ_INV_PKG_LOG_20210604AM075249.TPQS2I - SUCCESSFUL.
15. OOD: Performing Housekeeping - Remove Temporary Log File - /tmp/SBS_CPQ_INV_INF_CONC_SH.prog.21679 
15. OOD: Returning to Main Controlling Script

Successfully resubmitted concurrent program SBS_CPQ_INV_INF_CONC_SH with request ID 35389481 to start at 05-JUN-2021 05:30:00 (ROUTINE=AFPSRS)


+---------------------------------------------------------------------------+
No completion options were requested.

Output file size: 



Output is not being printed because:
The print option has been disabled for this report.

+---------------------------------------------------------------------------+
Concurrent request completed successfully
Current system time is 04-JUN-2021 07:54:21

+---------------------------------------------------------------------------+

**Comment by hamiltonmark** [comment]
This ticket can be closed.

**Comment by hamiltonmark** [comment]
Thanks Amit

1.       I tested - SBS CPQ File Extract Interface Inventory Items.
$APPLCSF value was added and job is successful now.
I added comment to close in the ticket.


2.       The next job to test after this one is - SBS CPQ File Extract Interface BOMS.
It ran successfully


3.       The next job to test after this one is - SBS CPQ File Extract Interface Price Lists.
I opened another ticket for this issue because it is failing.
Reason….
select *
from qp_list_headers_all_b
Returns - no table found... but...
select *
from apps.qp_list_headers_all_b
Returns successfully.
PACKAGE PBS.sbs_cpq_prclst_pkg is in error because of this issue.
